core/src/main/java/io/micronaut/core/io/service/ServiceScanner.java

```diff
@@ -234,6 +220,31 @@ private void findMicronautMetaServiceConfigs(BiConsumer<URI, String> consumer) t
                 consumer.accept(uri, path);
             }
         }
+        if (jrtProvider != null && jrtProvider.isOpen()) {
+            try {
+                jrtProvider.close();
+            } catch (Throwable ignore) {
+                // Ignore
+            }
+        }
+    }
+
+    @Nullable
+    private FileSystem getJrtProvider() {
+        try {
+            URI uri = URI.create("jrt:/");
+            FileSystem fs = FileSystems.getFileSystem(uri);
+            if (fs.isOpen()) {
+                return fs;
+            }
+            fs = FileSystems.newFileSystem(uri, Collections.emptyMap(), classLoader);
+            if (fs.isOpen()) {
+                return fs;
+            }
+        } catch (Exception e) {
```

Comment:
need to catch Throwable here, ServiceLoader.fail() throws ServiceConfigurationError

Reply:
BTW I wasn't able to reproduce it. Is there something special in your case?